openpype/lib/avalon_context.py

```diff
@@ -1971,3 +1971,123 @@ def get_last_workfile(
         return os.path.normpath(os.path.join(workdir, filename))
 
     return filename
+
+
+@with_avalon
+def get_linked_ids_for_representations(project, repre_ids, dbcon=None,
+                                       link_type=None, max_depth=0):
+    """Returns list of linked ids of particular type (if provided).
+
+    Goes from representations to version, back to representations
+    Args:
+        project (str)
+        repre_ids (list) or (ObjectId)
+        dbcon (avalon.mongodb.AvalonMongoDB, optional): Avalon Mongo connection
+            with Session.
+        link_type (str): ['reference', '..]
+        max_depth (int): limit how many levels of recursion
+    Returns:
+        (list) of ObjectId - linked representations
+    """
+    if not dbcon:
+        log.debug("Using `avalon.io` for query.")
+        dbcon = avalon.io
+        # Make sure is installed
+        dbcon.install()
+
+    if dbcon.Session["AVALON_PROJECT"] != project:
+        dbcon.Session["AVALON_PROJECT"] = project
+
+    if not isinstance(repre_ids, list):
+        repre_ids = [repre_ids]
+
+    versions = avalon.io.find(
+        {
+            "_id": {"$in": repre_ids},
+            "type": "representation"
+        },
+        projection={"parent": True}
+    )
+    version_ids = [version["parent"] for version in versions]
+
+    graph_lookup = {
+        "from": project,
+        "startWith": "$data.inputLinks.id",
+        "connectFromField": "data.inputLinks.id",
+        "connectToField": "_id",
+        "as": "outputs_recursive",
+        "depthField": "depth"
+    }
+    if max_depth != 0:
+        # We offset by -1 since 0 basically means no recursion
+        # but the recursion only happens after the initial lookup
+        # for outputs.
+        graph_lookup["maxDepth"] = max_depth - 1
+
+    match = {
+        "_id": {"$in": version_ids},
+        "type": "version"
+    }
+
+    pipeline_ = [
+        # Match
+        {"$match": match},
+        # Recursive graph lookup for inputs
+        {"$graphLookup": graph_lookup}
+    ]
+
+    result = dbcon.aggregate(pipeline_)
+    referenced_version_ids = _process_referenced_pipeline_result(result,
+                                                                 link_type)
+
+    representations = avalon.io.find(
+        {
+            "parent": {"$in": list(referenced_version_ids)},
+            "type": "representation"
+        },
+        projection={"_id": True}
+    )
+    ref_ids = {representation["_id"] for representation in representations}
```

Comment:
I believe this could also be:
```suggestion
    ref_ids = io.distinct("_id", 
        filter={
            "parent": {"$in": list(referenced_version_ids)},
            "type": "representation"
        }
    )
```
_Untested, so please consider as **pseudocode**_